clippy_lints/src/methods/parsed_string_literals/primitive_types.rs

```diff
@@ -0,0 +1,94 @@
+use std::borrow::Cow;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use clippy_utils::get_parent_expr;
+use clippy_utils::source::{snippet_with_applicability, str_literal_to_char_literal};
+use rustc_errors::Applicability;
+use rustc_hir::{self as hir, AmbigArg, Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_middle::ty::{self, Ty};
+use rustc_span::Symbol;
+
+pub(super) fn check(
+    cx: &LateContext<'_>,
+    expr: &Expr<'_>,
+    lit: Symbol,
+    ty: Ty<'_>,
+    strlit: &Expr<'_>,
+    explicit_type: Option<hir::Ty<'_, AmbigArg>>,
+) {
+    let mut app = Applicability::MachineApplicable;
+
+    macro_rules! number {
+        ($kind:ident, $expr:expr, $msg:expr, [$($subkind:ident => $ty:ident),*$(,)?]$(,)?) => {{
+            match $expr {
+                // If the right type has been found, return the string to substitute the parsing
+                // call with, which will be the literal followed by a suffix if initially the
+                // parse call was qualified with the return type. Note that we use a canonical
+                // suffix, whereas the parse call might have been qualified with a type alias,
+                // because type aliases can't be used as suffixes.
+                $(ty::$kind::$subkind =>
+                  (
+                      try_parse::<$ty>(
+                          lit,
+                          Some(stringify!($ty)),
+                          explicit_type.map(|ty| snippet_with_applicability(cx, ty.span, "_", &mut app))
```

Comment:
A snippet isn't needed here and it's the slowest available option. The fastest thing to do is match the type down to a single segment path and compare the symbol.